src/security-server/admin-service/application/src/main/java/org/niis/xroad/securityserver/restapi/repository/LocalGroupRepository.java

```diff
@@ -48,12 +50,19 @@
 public class LocalGroupRepository {
 
     private final PersistenceUtils persistenceUtils;
+    private final GroupMemberDAOImpl groupMemberDAO = new GroupMemberDAOImpl();
 
     public LocalGroupType getLocalGroup(Long entityId) {
         LocalGroupDAOImpl localGroupDAO = new LocalGroupDAOImpl();
         return localGroupDAO.getLocalGroup(persistenceUtils.getCurrentSession(), entityId);
     }
 
+    public void deleteGroupMembersByMemberId(ClientId.Conf memberId) {
+        var session = persistenceUtils.getCurrentSession();
+        groupMemberDAO.findAllByGroupMemberId(session, memberId)
+                .forEach(session::remove);
```

Comment:
This could be done with single delete statement instead of fetching and deleting in a loop